bluetooth: host: Some small CS fixes and improvements

olivier-le-sage opened the pull request:
1. Add docs on valid t_XX times for CS Test command
2. Fix values for minimizing/maximizing TX power
3. Add defines for RTT packet quality and RSSI for mode 1 steps
4. Make ToX-ToX values int16_t so the user doesn't need to cast
5. minor doc fix for bt_le_cs_step_data_parse()

jhedberg commented:
@olivier-le-sage when you have to enumerate multiple independent things, even though they're part of the same area/feature, it's a good indication that you might want to split the changes up into individual commits. In case there's a bug in one of the changes this then helps with bisectability. You also get a clearer mapping from the commit message to the actual change.

olivier-le-sage commented:
> @olivier-le-sage when you have to enumerate multiple independent things, even though they're part of the same area/feature, it's a good indication that you might want to split the changes up into individual commits. In case there's a bug in one of the changes this then helps with bisectability. You also get a clearer mapping from the commit message to the actual change.

sure, can split it up.

olivier-le-sage committed:
bluetooth: host: Fix values for minimizing/maximizing TX power

The values were swapped, leading to the opposite of the intended
behavior

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>

olivier-le-sage committed:
bluetooth: host: Add docs for t_XX times in CS Test

These options are in microseconds but not fully configurable,
document what options are allowed in 6.0

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>

olivier-le-sage committed:
bluetooth: host: uint16_t -> int16_t mode 1 deltas

These are signed 16-bit integers by spec

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>

olivier-le-sage committed:
bluetooth: host: Add defines for mode 1 packet quality and RSSI

Add some defines for packet quality and rssi for RTT,
these are useful for filtering out steps where the RTT exchange
was unsuccessful

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>

olivier-le-sage commented on include/zephyr/bluetooth/cs.h:
```diff
@@ -540,6 +590,35 @@ struct bt_le_cs_subevent_step {
+	int16_t i;
+	int16_t q;
+	struct {
+		int16_t x: 12;
```
an alternative is maybe

int16_t i = ((sample & BT_HCI_LE_CS_PCT_MASK) ^ BIT(11)) - BIT(11);
int16_t q = ((sample >> 12) ^ BIT(11)) - BIT(11);

jhedberg replied:
Both look quite hard to parse for me, however I'd at least add a code comment with a reference to the sign extension web page.

alwa-nordic replied:
Assigning a value to a signed integer variable that is larger than the variable can represent is technically implementation specific behavior in C.

It's probably OK and left implementation specific just in case the code runs a machine with non-twos complement integers. Maybe we can get a guarantee from Zephyr toolchain maintainers that this trick is supported?

Still, performing the conversion arithmetically avoids this caveat. I think it can be make quite clean:

```c
struct bt_le_cs_iq_sample bt_le_cs_parse_pct(const uint8_t pct[3])
{
    uint32_t sample = sys_get_le24(pct);

    /* Exract fields i and q. */
    uint16_t i2 = sample % (1<<12);
    uint16_t q2 = sample / (1<<12);

    /* Decode the fields from twos complement (bit width 12) into signed integers. */
    int16_t i = (i2 ^ (1<<11)) - (1<<11);
    int16_t q = (q2 ^ (1<<11)) - (1<<11);

    return (struct bt_le_cs_iq_sample){.i = i, .q = q};
}
```

olivier-le-sage replied:
decided to introduce this later in a separate PR

jhedberg commented on include/zephyr/bluetooth/cs.h:
```diff
@@ -540,6 +590,35 @@ struct bt_le_cs_subevent_step {
+ *
+ * @param pct 24-bit little-endian phase correction term.
+ *
+ * @return Real and imaginary terms as int16_t.
```
It's not returning `int16_t` though, rather `struct bt_le_cs_iq_sample`.

jhedberg replied:
> It's not returning `int16_t` though, rather `struct bt_le_cs_iq_sample`.

Nevermind - it's just a reference to the implementation of `structure bt_le_cs_iq_sample`. It's just not clear from the text alone that it contains two `int16_t` parts.

olivier-le-sage replied:
can rephrase to `@return bt_le_cs_iq_sample struct containing real and imaginary terms as int16_t`

jhedberg replied:
Sure

olivier-le-sage committed:
bluetooth: host: minor doc correction for bt_le_cs_step_data_parse()

This function can be used to parse HCI-formatted subevent results
regardless of whether they come from CS Test or not

Signed-off-by: Olivier Lesage <olivier.lesage@nordicsemi.no>
